### DISCUSSION TITLE
How to get the latest Commit from a Pull request?

### DISCUSSION THREAD

**Discussion by Andre601** [community/community#26676] [Actions]
<p>My goal is to get the latest commit made in a Pull request using a <code>pull_request_target</code> workflow run.</p>
<p>I tried using <code>${{ github.sha }}</code> because according to the docs should it return “the SHA of the commit that triggered this workflow” yet the returned commit sha seems to be from the target branch and not the branch that should be merged into the target.</p>
<p>What other ways would there be?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>See the <a href="https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request">documentation for the <code>pull_request</code> event</a>:</p>
<blockquote>
<p>Note that <code>GITHUB_SHA</code> for this event is the last merge commit of the pull request merge branch. If you want to get the commit ID for the last commit to the head branch of the pull request, use <code>github.event.pull_request.head.sha</code> instead.</p>
</blockquote>

**Comment by Andre601** [comment]
<p>I tried with <code>${GITHUB_SHA}</code> but for some reason is it not turned into the actual SHA.</p>
<p>Idk why this happens but I had this many times before on other workflows and never found a solution nor cause for it.<br>
Also, to be clear: I tried <code>${GITHUB_SHA}</code> but also <code>{GITHUB_SHA}</code> and also with spaces but no changes.</p>

**Comment by airtower-luna** [comment]
<p>First of, the paragraph I quoted says that <code>GITHUB_SHA</code> (and thus <code>github.sha</code>, it’s the same, just as an Actions variable, not shell) is <em>not</em> what you want, and it says what to use instead. Please read it again. :slightly_smiling_face:</p>
<p>Secondly, <code>${NAME}</code> is shell syntax, so whichever variable you use it’s only going to work in a shell context (e.g. a <code>run</code> step using Bash).</p>

**Comment by Andre601** [comment]
<p>Okay, yeah. I missread that, so my bad.</p>
<p>I figured this out a little later myself, so it works now like I want it to.</p>
